index.bs

```diff
@@ -577,66 +577,155 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
+		Further, apps are expected to enumerate the mime-types of the [=ClipboardItem=] they are pasting and select the one best-suited for the app according to some app-specific algorithm.
+		Alternatively, an app can present the user with options on how to paste a [=ClipboardItem=], e.g. “paste as image” or “paste formatted text”, etc.
+
+		A [=ClipboardItem=] contains multiple representations. Each representation consists of a MIME type and a corresponding {{DOMString}} or [=Blob=].
+
+		<dl class=note>
+			<dt><code><var>clipboardItem</var> = new ClipboardItem([<var>items</var>, <var>options</var>])</code>
+			<dd>
+			Creates a new [=ClipboardItem=] object. <var>items</var> are used to fill its MIME types and [=Promise=]s to [=Blob=]s corresponding to the MIME types, <var>options</var> can be used to fill its {{ClipboardItemOptions}},
+			as per the example below.
+
+			<pre class="example javascript">
+			const format1 = 'text/plain';
+			const promise_text_blob = Promise.resolve(new Blob(['hello'], {type: format1}));
+			const clipboardItemInput = new ClipboardItem(
+				{[format1]: promise_text_blob},
+				{options: "unspecified"});
+			</pre>
+
+			<dt><code><var>clipboardItem</var>.getType(<var>type</var>)</code>
+			<dd><p>Returns a [=Promise=] to the [=Blob=] corresponding to <var>type</var>.</p>
 
-		enum PresentationStyle { "unspecified", "inline", "attachment" };
+			<dt><code><var>clipboardItem</var>.<var>types</var></code>
+			<dd><p>Returns the list of <var>types</var> contained in the <var>clipboardItem</var> object.
 
-		dictionary ClipboardItemOptions {
-		  PresentationStyle presentationStyle = "unspecified";
+		</dl>
+		<p>
+			The <a constructor lt="ClipboardItem()">ClipboardItem</a> step for <code>new ClipboardItem(<var>items</var>, <var>options</var>)</code> is to set [=this=]'s items to <var>items</var> and options to <var>options</var>
+		</p>
+
+		<h4 attribute for=ClipboardItem lt=presentationStyle>presentationStyle</h4>
+		<p>It is an <a href=https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#enumerated-attribute>enumerated attribute</a> whose keywords are the <code>string</code>, <code>unspecified</code>, <code>inline</code> and <code>attachment</code>.
+		
+		Note: {{ClipboardItem/presentationStyle}} helps distinguish between "inline" data(e.g. selecting text on a web page and copying), vs. file-like data (e.g. copying a plain text file).
+				  This difference is used to provide a hint when writing to the system pasteboard on iOS/iPadOS. This allows apps like Notes to insert a file attachment when copying a plain text file from the Files app, but insert text inline when
+				  copying a selected piece of plain text from another app. In both cases, the MIME type in the pasteboard would be text/plain. This is used for both reading and writing.
+		</p>
+
+		<p>{{ClipboardItem/presentationStyle}} getter steps are to return [=this=]'s {{ClipboardItem/presentationStyle}}</p>
```

Comment:
This definition is circular; it is returning itself.

You need to instead define an associated concept, and have the getter return that. E.g.

```html
Each {{ClipboardItem}} has a <dfn for="ClipboardItem">presentation style</dfn>, which is a {{PresentationStyle}}.

The <dfn attribute for="ClipboardItem">presentationStyle</dfn> getter steps are to return [=this=]'s [=ClipboardItem/presentation style=].
```

Then, somewhere else in the spec, you need to set a given ClipboardItem's `[=ClipboardItem/presentation style=]`.

Reply:
Done.